From a5088f05e575581d5917c3f26cc08b3b52bc1637 Mon Sep 17 00:00:00 2001
From: Igor Grinberg <igrinberg@habana.ai>
Date: Tue, 03 Aug 2021 16:20:24 +0300
Subject: [PATCH] [SW-51026] gaudi: security: fix mme protect

The protection bits setting for MME ACC and SBAB were not configured
correctly due to a bug in base calculation.
base variable must be calculated as part of the MME walk over loop and
not outside it once.
Fix protection bits setting by fixing the base calculation.

Change-Id: Ie94d8c7de8bc5c407ca677beb5e8617ed531f7fe
Signed-off-by: Igor Grinberg <igrinberg@habana.ai>
---

diff --git a/arch/gaudi/security.c b/arch/gaudi/security.c
index 09d1fbc..542542c 100644
--- a/arch/gaudi/security.c
+++ b/arch/gaudi/security.c
@@ -934,16 +934,17 @@
 
 static void preboot_secure_mme_acc(void)
 {
-	u32 reg, mask, prot_base, base = mmMME0_ACC_BASE & ~HL_CFG_BASE;
+	u32 reg, mask, prot_base, base;
 	int i;
 
 	for (i = 0; i < MMES_NUM; i++) {
+		base = (mmMME0_ACC_BASE & ~HL_CFG_BASE) + MMES_DIST * i;
 		/* Set ACC to user */
 		reg = mmMME0_ACC_PROT + MMES_DIST * i;
 		regmap_writel(PROT_OVR_USER, reg);
 
 		/* blocking regs within offsets: 0 - 0x80 */
-		prot_base = base + MMES_DIST * i;
+		prot_base = base;
 		mask = ~(PROT_BIT(mmMME0_ACC_ACC_STALL) &
 			 PROT_BIT(mmMME0_ACC_WBC));
 		regmap_writel(mask, PRIV_PROT_REG(prot_base));
@@ -956,16 +957,17 @@
 
 static void preboot_secure_mme_sbab(void)
 {
-	u32 reg, mask, prot_base, base = mmMME0_SBAB_BASE & ~HL_CFG_BASE;
+	u32 reg, mask, prot_base, base;
 	int i;
 
 	for (i = 0; i < MMES_NUM; i++) {
+		base = (mmMME0_SBAB_BASE & ~HL_CFG_BASE) + MMES_DIST * i;
 		/* Set SBAB to user */
 		reg = mmMME0_SBAB_PROT + MMES_DIST * i;
 		regmap_writel(PROT_OVR_USER, reg);
 
 		/* blocking regs within offsets: 0 - 0x80 */
-		prot_base = base + MMES_DIST * i;
+		prot_base = base;
 		mask = ~(PROT_BIT(mmMME0_SBAB_SB_STALL) &
 			 PROT_BIT(mmMME0_SBAB_ARUSER0) &
 			 PROT_BIT(mmMME0_SBAB_ARUSER1));
